src/helpers/apiInteraction.js

```diff
@@ -0,0 +1,35 @@
+// Common class for interacting with APIs
+export class ApiInteraction {
+
+    constructor() {
+        this.url = 'http://localhost:5000/';
+    }
+
+    async getProducts() {
+        let data = await fetch(`${this.url}products`);
+        data = await data.json();
+        return data;
+    }
+
+    async getCategories() {
+        let data = await fetch(`${this.url}categories`);
+        data = await data.json();
+        return data;
+    }
+
+    async getBanners() {
+        let data = await fetch(`${this.url}banners`);
+        data = await data.json();
+        return data;
+    }
+
+    async addToCart(id) {
+        let data = await fetch(`${this.url}addToCart/`, {
+            method: "POST",
+            body: JSON.stringify({ productId: id })
+        });
+        data = await data.json();
+        return data;
```

Comment:
Over-riding the `data` variable won't be necessary. you can directly return ...data.json(). Since, there would be no overrides use `const` for declaring variable.